framework/CMakeLists.txt

```diff
@@ -101,8 +101,12 @@ include(multiap-helpers)
 option(BUILD_EXAMPLES "build examples" OFF)
 option(BUILD_TESTS "build multiap unit tests" OFF)
 option(PASSIVE_MODE "build framework passive mode" OFF)
+
+set(MSGLIB "zmq" CACHE STRING "Which messaging library backend to use")
+set_property(CACHE MSGLIB PROPERTY STRINGS "zmq" "nng" "None")
 if(PASSIVE_MODE)
-unset(MSGLIB)
+message(STATUS "Force MSGLIB=None in passive mode")
+set(MSGLIB "None")
```

Comment:
You could use a construct like

```cmake
if(PASSIVE_MODE)
set(MSGLIB "None" INTERNAL)
else()
set(MSGLIB "zmq" CACHE STRING "Which messaging library backend to use")
set_property(CACHE MSGLIB PROPERTY STRINGS "zmq" "nng" "None")
endif()
```

However, I don't know if that works properly with cmake-gui, or even at all. It might be necessary to make it an INTERNAL cache variable also in the PASSIVE_MODE case.

Reply:
This doesn't work, if I do it the way you wrote, the "INTERNAL" gets into the string then `find_package` fails':
```
- Done                                                                                                                                                                                                     
-- Messaging library - None;INTERNAL                                                                                                                                                                        
CMake Error at common/CMakeLists.txt:19 (find_package):                                                                                                                                                     
  find_package called with invalid argument "INTERNAL"       
```

If i add cache it also doesn't solve it. I will leave it as I did it, but fix the common/test/CMakeLists.txt